common/views/components/PaletteColorPicker/HueSlider.tsx

```diff
@@ -0,0 +1,176 @@
+import styled from 'styled-components';
+import React, {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+
+type Props = {
+  hue: number;
+  onChangeHue: (value: number) => void;
+};
+
+const HueBar = styled.div`
+  position: relative;
+  width: 100%;
+  height: 28px;
+  background: linear-gradient(
+    to right,
+    #f00 0%,
+    #ff0 17%,
+    #0f0 33%,
+    #0ff 50%,
+    #00f 67%,
+    #f0f 83%,
+    #f00 100%
+  );
+`;
+
+type HandleProps = {
+  leftOffset: number;
+};
+
+const Handle = styled.span.attrs<HandleProps>(({ leftOffset }) => ({
+  style: {
+    left: `${leftOffset * 100}%`,
+  },
+}))<HandleProps>`
+  display: inline-block;
+  position: absolute;
+  top: 50%;
+  width: 6px;
+  height: 24px;
+  transform: translate(-50%, -50%);
+  background-color: white;
+  box-shadow: 0 0 1px rgba(black, 0.5);
+  border-radius: 2px;
+`;
+
+type InteractionEvent = MouseEvent | TouchEvent;
+type ReactInteractionEvent = React.MouseEvent | React.TouchEvent;
+
+const isTouch = (e: InteractionEvent): e is TouchEvent => 'touches' in e;
+const clamp = (x: number, min = 0, max = 1) =>
```

Comment:
nit: We have `lodash.clamp`.

Reply:
Do we? I didn't want to add a dependency for such a simple function

Reply:
Yea, we could put it in `utils/`